src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -284,11 +282,16 @@ void sendAsync(ProducerRecord<String, String> record, Callback callback) {
 
 			// javascript topic override
 			topic = r.getKafkaTopic();
-			if ( topic == null ) {
+			if (topic == null) {
 				topic = generateTopic(this.topic, pk);
 			}
+			LOGGER.debug("context.getConfig().producerPartitionKey = " + context.getConfig().producerPartitionKey);
 
-			record = new ProducerRecord<>(topic, this.partitioner.kafkaPartition(r, getNumPartitions(topic)), key, value);
+			if ("random".equals(context.getConfig().producerPartitionKey)) {
+				LOGGER.debug("**********take effect***********");
+				record = new ProducerRecord<>(topic, value);
+			} else
+				record = new ProducerRecord<>(topic, this.partitioner.kafkaPartition(r, getNumPartitions(topic)), key, value);
```

Comment:
i don't really see how this produces random distribution...

Reply:
Isn't kafka's default partitioning rule polling when the key is empty?

Reply:
![image](https://user-images.githubusercontent.com/18672877/76213781-603c0380-6246-11ea-96dc-8e8df38024ce.png)

Reply:
Hello @osheroff,The code is formatted